test/parallel/test-timers-same-timeout-wrong-list-deleted.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+/*
+ * This is a regression test for https://github.com/nodejs/node/issues/7722.
+ *
+ * When nested timers have the same timeout, calling clearTimeout on the
+ * older timer after it has fired causes the list the newer timer is in
+ * to be deleted. Since the newer timer was not cleared, it still blocks
+ * the event loop completing for the duration of its timeout, however, since
+ * no reference exists to it in its list, it cannot be canceled and its
+ * callback is not called when the timeout elapses.
+ */
+
+require('../common');
+const assert = require('assert');
+const Timer = process.binding('timer_wrap').Timer;
+
+const TIMEOUT = 100;
+const start = Timer.now();
+
+function testNestedTimers() {
+  // This bug also prevents the erroneously dereferenced timer's callback
+  // from being called, so we can't use it's execution or lack thereof
+  // to assert that the bug is fixed.
+  process.on('exit', function() {
+    const end = Timer.now();
+    assert.equal(end - start < TIMEOUT * 2, true,
+                 'Elapsed time does not include second timer\'s timeout.');
+  });
+
+  const handle1 = setTimeout(function() {
+    // Cause the old TIMEOUT list to be deleted
+    clearTimeout(handle1);
+
+    // Cause a new list with the same key (TIMEOUT) to be created for this timer
+    const handle2 = setTimeout(function() {
+      assert.fail(true, false, 'Inner callback is not called');
+    }, TIMEOUT);
+
+    setTimeout(function() {
```

Comment:
This callback should be wrapped in `common.mustCall` so that we make sure that `clearTimeout(handle2)` is called. Not doing so might cause false negatives where the test would pass because `handle2` wasn't cleared due to another timers bug, but the bug that this change fixes could still be present (or be reintroduced).